lib/index.js

```diff
@@ -0,0 +1,109 @@
+/*!
+ * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const assert = require('assert-plus');
+const bedrock = require('bedrock');
+const {util: {clone, BedrockError}} = bedrock;
+
+const packages = [];
+
+/**
+ * @module bedrock-package-manager
+ */
+
+/**
+ * Get package information. One of `alias` or `packageName` is required.
+ *
+ * @param {Object} options - The options to use.
+ * @param {string} options.type - The package type.
+ * @param {string} [options.alias] - The package alias.
+ * @param {string} [options.packageName] - The package name.
+ *
+ * @returns {PackageInfo} The package information.
+ * @throws {BedrockError} Will throw a `NotFoundError` if the package is not
+ *   found.
+ */
+exports.get = ({alias, packageName, type}) => {
+  assert.optionalString(alias, 'options.alias');
+  assert.optionalString(packageName, 'options.packageName');
+  assert.string(type, 'options.type');
+
+  let pkg;
+  if(alias) {
+    pkg = packages.find(p => (p.type === type && p.alias === alias));
+  }
+  if(packageName) {
+    pkg = packages.find(p => (p.type === type &&
+      p.packageName === packageName));
+  }
+
+  if(!pkg) {
+    throw new BedrockError('Package not found.', 'NotFoundError', {
+      alias,
+      httpStatusCode: 404,
+      packageName,
+      public: true,
+      type,
+    });
+  }
+  return clone(pkg);
+};
+
+/**
+ * Find packages by type.
+ *
+ * @param {Object} options - The options to use.
+ * @param {string} options.type - The package type to find.
+ *
+ * @returns {PackageInfo[]} The package information.
+ */
+exports.findType = ({type}) => {
+  assert.string(type, 'options.type');
+
+  return clone(packages.filter(p => p.type === type));
+};
+
+/**
+ * Register a package.
+ *
+ * @param {Object} options - The options to use.
+ * @param {string} options.packageName - The NPM package name.
+ * @param {string} options.type - The package type.
+ * @param {Object} [options.meta={}] - Domain specific meta data.
+ * @param {string} [options.alias] - The package alias. A short name for the
+ *   package (e.g. 'mongodb', 'redis').
+ *
+ * @returns {PackageInfo[]} The package information.
+ * @throws {BedrockError} Will throw a `DuplicateError` if the package is
+ *   already registered.
+ */
+exports.register = ({alias, meta = {}, packageName, type}) => {
+  assert.optionalString(alias, 'options.alias');
+  assert.string(packageName, 'options.name');
+  assert.string(type, 'options.type');
+  assert.object(meta, 'options.meta');
+
+  if(packages.find(p => (p.packageName === packageName && p.type === type))) {
```

Comment:
We also need to check `alias` + `type` right?

Reply:
That's a good idea, I'll fix it up.

Reply:
Done.